src/App.js

```diff
@@ -1,17 +1,56 @@
+// import the required libraries
 import React from 'react';
 import { BrowserRouter as Router, Route, Switch } from 'react-router-dom';
+
+// import the pages/components
+import { Provider } from 'react-redux';
+import { createStore, applyMiddleware, compose } from 'redux';
+import thunk from 'redux-thunk'; // import thunk middleware
+import createSagaMiddleware from '@redux-saga/core';
 import { LandingPage } from './components/landingPage/components';
-import LoginPage from './components/login/components';
+import { components as LoginPage } from './components/login';
 import AdminLoginPage from './components/adminLoginPage/components';
 
+// import the combinedSagas and the combinedReducers
```

Comment:
why are you using `redux thunk` and `redux-sagas`

Reply:
i will be performing some async operations for authentication

Reply:
`redux sagas` can handle that for you

Reply:
yes, that is what i use, i also use redux-thunk in order to call my actions as functions instead of passing an object

Reply:
@Xand6r do not use `redux-think`, handle all of your async operation with `redux-sagas`

Reply:
@enyeInc all my async operations are performed by redux-sagas. I use redux thunk in order to dispatch functions which are synchronous, as opposed to dispatching objects wothout redux-thunk. i dispatch functions so i handle data cleansing and formatting in each action(which is a function) rather than directly in the component

Reply:
@Xand6r Please do not do that, pass the object to the sagas and do your data cleansing in the saga function. DO NOT mix thunks with sagas.

Reply:
i will do this. thank you @enyeInc